test/forcing/GridDataSelector_Test.cpp

```diff
@@ -0,0 +1,200 @@
+            }
+
+            cell.value = values_[cell.x + (cell.y * spec_.columns)];
+        }
+
+        return result;
+    }
+
+    static const SelectorConfig default_selector;
+
+  private:
+
+    void initialize_() noexcept {
+        values_.resize(spec_.columns * spec_.rows);
+        for (size_t i = 0; i < spec_.rows; ++i) {
+            for (size_t j = 0; j < spec_.columns; ++j) {
+                values_.at(j + (i * spec_.columns)) = static_cast<double>(i + j);
+            }
+        }
+    }
+
+    static const std::string variable_;
+    GridSpecification spec_;
+    std::vector<double> values_;
+};
+
+const std::string TestGridDataProvider::variable_ = "variable";
+const SelectorConfig TestGridDataProvider::default_selector = {
+    0, // init_time
+    3599, // duration
+    "variable", // variable
+    "m", // units
+};
+
+// x is column
+// y is row
+inline Cell make_cell_xy(std::uint64_t x, std::uint64_t y)
+{ return { x, y, static_cast<uint64_t>(-1), NAN}; }
+
+inline constexpr geojson::coordinate_t make_point(double x, double y)
+{ return { x, y }; }
+
+// Tests for individual cell selection, providing the exact cells
+// we want to pull from the gridded data provider. Checks that
+// the number of cells returned matches the number of requests,
+// and that the indices and values are as expected.
+TEST(GridDataSelectorTest, CellSelection) {
+    TestGridDataProvider provider{};
+    GridDataSelector selector{
+        TestGridDataProvider::default_selector,
+        {{ // cells
+            make_cell_xy(0, 0),
+            make_cell_xy(5, 2),
+            make_cell_xy(9, 9)
+        }}
+    };
+    
+    const auto cells = provider.get_values(selector, data_access::ReSampleMethod::SUM);
+
+    const auto expect_cell = [&cells](std::size_t index, std::uint64_t x, std::uint64_t y)
+    {
+        const auto& cell = cells[index];
+        EXPECT_EQ(cell.x, x);
+        EXPECT_EQ(cell.y, y);
+        EXPECT_EQ(cell.z, static_cast<uint64_t>(-1));
+        EXPECT_EQ(cell.value, static_cast<double>(x + y));
+    };
+    
+    ASSERT_EQ(cells.size(), 3);
+    expect_cell(/*index=*/ 0, /*x=*/ 0, /*y=*/ 0);
+    expect_cell(/*index=*/ 1, /*x=*/ 5, /*y=*/ 2);
+    expect_cell(/*index=*/ 2, /*x=*/ 9, /*y=*/ 9);
+}
+
+// Tests for extent-based selection using a bounding box
+// for a subset of the overarching grid.
+TEST(GridDataSelectorTest, ExtentSelection) {
+
+    GridSpecification grid_spec {
+        10, // rows 
+        10, // cols
+        {20, 30, 20, 30}
+    };
+
+    TestGridDataProvider provider{grid_spec};
+
+    // Only take the upper-right 25 cells.
+    GridDataSelector selector{
+        TestGridDataProvider::default_selector,
+        grid_spec,
+        { 25, 30, 25, 30}
+    };
+
+    const auto cells = provider.get_values(selector, data_access::ReSampleMethod::SUM);
+    ASSERT_GT(cells.size(), 0);
+    EXPECT_EQ(cells.size(), 25);
```

Comment:
```suggestion
    EXPECT_EQ(cells.size(), 25);
```
That's kinda redundant with an exact expected value.

Reply:
Removed in [c3fb2fa](https://github.com/NOAA-OWP/ngen/pull/803/commits/c3fb2fae5ca7bb87192d2bb0b59b2c4dd016e6e6)